resources/views/people/_header.blade.php

```diff
@@ -27,10 +27,14 @@
 
           <h2 class="{{ \App\Helpers\LocaleHelper::getDirection() }}">
             {{ $contact->getCompleteName(auth()->user()->name_order) }}
-            @if ($contact->birthday_special_date_id)
+            @if ($contact->birthday_special_date_id && !($contact->is_dead))
               @if ($contact->birthdate->getAge())
                 <span class="ml3 light-silver f4">(<i class="fa fa-birthday-cake mr1"></i> {{ $contact->birthdate->getAge() }})</span>
               @endif
+            @elseif ($contact->is_dead)
+                @if (! is_null($contact->deceasedDate))
+                  <span class="ml3 light-silver f4">(Age at Death {{ $contact->deceasedDate->getDeathAge() }})</span>
```

Comment:
Multiple comments there:
- don't forget to add new translation keys instead of writing plain English in Blade files (by adding your string to `en/people.php` for instance).
- it's not obvious that you need to call the deceased date on the contact object to find out what the age at death was. I think it would be better to call a method on the contact itself, like `$contact->getAgeAtDeath()`. Then, this method can call a new method in the SpecialDate class to do the trick. That way, the Contact object doesn't care how it should get this age. It just knows that it has a method to do it for him.
- However, if you still want to use something like `$contact->deceasedDate->`, then you need another method, more generic, instead of the one you wrote. Carbon provides a `diffInYears` helper that you can probably use, like `$contact->deceasedDate->diffInYears($contact->birthdate)` or something.

The last two options work and I have no real preference for any of them.